utils/compression/zstd_compressor.go

```diff
@@ -0,0 +1,51 @@
+// Copyright (C) 2019-2023, Ava Labs, Inc. All rights reserved.
+// See the file LICENSE for licensing terms.
+
+package compression
+
+import (
+	"bytes"
+	"fmt"
+	"io"
+
+	"github.com/DataDog/zstd"
+)
+
+var _ Compressor = (*zstdCompressor)(nil)
+
+func NewZstdCompressor(maxSize int64) Compressor {
+	return &zstdCompressor{
+		maxSize: maxSize,
+	}
+}
+
+type zstdCompressor struct {
+	maxSize int64
+	reader  bytes.Reader
+}
+
+func (z *zstdCompressor) Compress(msg []byte) ([]byte, error) {
+	if int64(len(msg)) > z.maxSize {
+		return nil, fmt.Errorf("%w: (%d) > (%d)", ErrMsgTooLarge, len(msg), z.maxSize)
+	}
+	return zstd.Compress(nil, msg)
+}
+
+func (z *zstdCompressor) Decompress(msg []byte) ([]byte, error) {
+	z.reader.Reset(msg)
```

Comment:
This isn't safe for concurrent access. tbh I don't understand why gzip is done the way it is... having the reader be a local var makes much more sense imo (it will likely be allocated on the stack anyways)